src/test/java/com/passbook/sparkeighteen/controller/TransactionControllerTest.java

```diff
@@ -0,0 +1,122 @@
+
+    @InjectMocks
+    private TransactionController transactionController;
+
+    @Test
+    public void validRequest_successfulResponse_CreditTransactionSuccessfull() throws Exception {
+
+        Integer txnID = 1;
+
+        TransactionRequest transactionRequest = TransactionRequest.builder()
+                .amount(100F)
+                .name("ketan")
+                .note("Mobile Recharge")
+                .transactionType(TransactionType.CREDIT)
+                .build();
+
+        Mockito.when(transactionService.transact(txnID,transactionRequest)).thenReturn(TransactionResponse.builder()
+                .message("Credited")
+                .amount(100f)
+                .note("tea")
+                .transactionType(TransactionType.CREDIT)
+                .closingBalance(100f)
+                .build());
+
+        ResponseEntity<TransactionResponse> response = transactionController.transact(txnID,transactionRequest);
+        assertEquals(response.getStatusCode(), HttpStatus.OK);
+
+    }
+
+    @Test
+    public void InvalidRequest_errorResponse_CreditTransactionUnSuccessfull() throws Exception {
+
+        Integer txnID = 1;
+
+        TransactionRequest transactionRequest = TransactionRequest.builder()
+                .amount(-100f)
+                .transactionType(TransactionType.CREDIT)
+                .build();
+
+        Mockito.when(transactionService.transact(txnID,transactionRequest)).thenReturn(TransactionResponse.builder()
+                .message("amount cannot be less than zero")
+                .amount(null)
+                .note(null)
+                .transactionType(TransactionType.CREDIT)
+                .closingBalance(null)
+                .build());
+
+        ResponseEntity<TransactionResponse> response = transactionController.transact(txnID,transactionRequest);
+        assertEquals(response.getStatusCode(), HttpStatus.OK);
+
+    }
+
+    @Test
+    public void validRequest_successfulResponse_DebitTransactionSuccessfull() throws Exception {
+
+        Integer txnID = 1;
+
+        TransactionRequest transactionRequest = TransactionRequest.builder()
+                .amount(100F)
+                .name("Ashish")
+                .note("Wifi Recharge")
+                .transactionType(TransactionType.DEBIT)
+                .build();
+
+        Mockito.when(transactionService.transact(txnID,transactionRequest)).thenReturn(TransactionResponse.builder()
+                .message("Debited")
+                .amount(100f)
+                .note("coffee")
+                .transactionType(TransactionType.DEBIT)
+                .closingBalance(0f)
+                .build());
+
+        ResponseEntity<TransactionResponse> response = transactionController.transact(txnID,transactionRequest);
+        assertEquals(response.getStatusCode(), HttpStatus.OK);
+
+    }
+
+    @Test
+    public void InvalidRequest_errorResponse_DebitTransactionUnSuccessfull() throws Exception {
```

Comment:
naming convention ? (methodname_senario_result)

Reply:
Depends on individual, readable and understood by others just by taking a look.